lib/sqlite3/fork_safety.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require "weakref"
+
+# based on Rails's active_support/fork_tracker.rb
+module SQLite3
+  module ForkSafety
+    module CoreExt
+      def _fork
+        pid = super
+        if pid == 0
+          ForkSafety.discard
+        end
+        pid
+      end
+    end
+
+    @databases = []
+
+    class << self
+      def hook!
+        ::Process.singleton_class.prepend(CoreExt)
+      end
+
+      def track(database)
+        @databases << WeakRef.new(database)
```

Comment:
Do we need to make this thread-safe?

Reply:
Or should I just use `ObjectSpace.each_object(Database)`?

Reply:
Made it thread-safe.

Reply:
> Do we need to make this thread-safe?

🤷🏻‍♀️

If we're really worried about it, we could use a Queue object. Then we don't need to pull in another library.

Reply:
> Or should I just use `ObjectSpace.each_object(Database)` instead of keeping our own list?

I like the list better than iterating the heap.